src/kernels/kernelsum.jl

```diff
@@ -1,63 +1,58 @@
-    kernelmatrix(k, X) == kernelmatrix(k1 + k2, X)
-    kweighted = 0.5* k1 + 2.0*k2 == KernelSum([k1, k2], weights = [0.5, 2.0])
+Create a sum of kernels. One can also use the operator `+`
+
+```jldoctest
+julia> k1 = SqExponentialKernel();
+
+julia> k2 = LinearKernel();
+
+julia> KernelSum(k1, k2) == k1 + k2
+true
+
+julia> kernelmatrix(k, X) == kernelmatrix(k1, X) .+ kernelmatrix(k2, X)
+true
+
+julia> kernelmatrix(k, X) == kernelmatrix(k1 + k2, X)
+true
+
+julia> KernelSum((k1, k2)) == k1 + k2
+true
+
+julia> KernelSum([k1, k2)]) == k1 + k2
+false
+
+julia> KernelSum([k1, k2)]) == KernelSum((k1, k2)) 
+false
 ```
 """
-struct KernelSum <: Kernel
-    kernels::Vector{Kernel}
-    weights::Vector{Real}
+struct KernelSum{Tk} <: Kernel
+    kernels::Tk
 end
 
-function KernelSum(
-    kernels::AbstractVector{<:Kernel};
-    weights::AbstractVector{<:Real} = ones(Float64, length(kernels)),
-)
-    @assert length(kernels) == length(weights) "Weights and kernel vector should be of the same length"
-    @assert all(weights .>= 0) "All weights should be positive"
-    return KernelSum(kernels, weights)
+function KernelSum(kernel::Kernel, kernels::Kernel...)
+    return KernelSum((kernel, kernels...))
 end
 
-Base.:+(k1::Kernel, k2::Kernel) = KernelSum([k1, k2], weights = [1.0, 1.0])
-Base.:+(k1::ScaledKernel, k2::ScaledKernel) = KernelSum([kernel(k1), kernel(k2)], weights = [first(k1.σ²), first(k2.σ²)])
-Base.:+(k1::KernelSum, k2::KernelSum) =
-    KernelSum(vcat(k1.kernels, k2.kernels), weights = vcat(k1.weights, k2.weights))
-Base.:+(k::Kernel, ks::KernelSum) =
-    KernelSum(vcat(k, ks.kernels), weights = vcat(1.0, ks.weights))
-Base.:+(k::ScaledKernel, ks::KernelSum) =
-        KernelSum(vcat(kernel(k), ks.kernels), weights = vcat(first(k.σ²), ks.weights))
-Base.:+(k::ScaledKernel, ks::Kernel) =
-        KernelSum(vcat(kernel(k), ks), weights = vcat(first(k.σ²), 1.0))
-Base.:+(ks::KernelSum, k::Kernel) =
-    KernelSum(vcat(ks.kernels, k), weights = vcat(ks.weights, 1.0))
-Base.:+(ks::KernelSum, k::ScaledKernel) =
-        KernelSum(vcat(ks.kernels, kernel(k)), weights = vcat(ks.weights, first(k.σ²)))
-Base.:+(ks::Kernel, k::ScaledKernel) =
-        KernelSum(vcat(ks, kernel(k)), weights = vcat(1.0, first(k.σ²)))
-Base.:*(w::Real, k::KernelSum) = KernelSum(k.kernels, weights = w * k.weights) #TODO add tests
+Base.:+(k1::Kernel, k2::Kernel) = KernelSum(k1, k2)
+Base.:+(k1::KernelSum, k2::KernelSum) = KernelSum(k1.kernels..., k2.kernels...)
+Base.:+(k::Kernel, ks::KernelSum) = KernelSum(k, ks.kernels...)
 
 Base.length(k::KernelSum) = length(k.kernels)
 
-(κ::KernelSum)(x, y) = sum(κ.weights[i] * κ.kernels[i](x, y) for i in 1:length(κ))
+(κ::KernelSum)(x, y) = sum(κ.kernels[i](x, y) for i in 1:length(κ))
 
 function kernelmatrix(κ::KernelSum, x::AbstractVector)
-    return sum(κ.weights[i] * kernelmatrix(κ.kernels[i], x) for i in 1:length(κ))
+    return sum(kernelmatrix(κ.kernels[i], x) for i in 1:length(κ))
```

Comment:
IMO it would be easier to just iterate over the kernels, i.e.
```suggestion
    return sum(kernelmatrix(k, x) for k in κ.kernels)
```